sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/EditContactSteps.java

```diff
@@ -112,50 +255,349 @@ public EditContactSteps(WebDriverHelpers webDriverHelpers) {
+    webDriverHelpers.clickWebElementByText(MALIGNANCY_OPTIONS, text);
+  }
+
+  public void selectPreexistingConditionChronicPulmonary(String text) {
+    webDriverHelpers.clickWebElementByText(PULMONARY_OPTIONS, text);
+  }
+
+  public void selectPreexistingConditionRenal(String text) {
+    webDriverHelpers.clickWebElementByText(RENAL_OPTIONS, text);
+  }
+
+  public void selectPreexistingConditionNeurologic(String text) {
+    webDriverHelpers.clickWebElementByText(NEUROLOGIC_OPTIONS, text);
+  }
+
+  public void selectPreexistingConditionCardiovascular(String text) {
+    webDriverHelpers.clickWebElementByText(CARDIOVASCULAR_OPTIONS, text);
+  }
+
+  public void fillAdditionalRelevantPreexistingConditions(String source) {
+    webDriverHelpers.clearAndFillInWebElement(ADDITIONAL_RELEVANT_PRE_CONDITIONS_TEXT, source);
+  }
+
+  public void selectVaccinationStatusForThisDisease(String vaccine) {
+    webDriverHelpers.selectFromCombobox(VACCINATION_STATUS_COMBOBOX, vaccine);
+  }
+
+  public void selectImmunosuppressiveTherapy(String text) {
+    webDriverHelpers.clickWebElementByText(IMMUNOSUPPRESSIVE_THERAPY_OPTIONS, text);
+  }
+
+  public void selectActiveInCare(String text) {
+    webDriverHelpers.clickWebElementByText(CARE_OVER_60_OPTIONS, text);
+  }
+
+  public void clickCancelFollowUpButton() {
+    if (editedContact.isCancelFollowUp()) {
+      webDriverHelpers.clickOnWebElementBySelector(CANCEL_FOLLOW_UP_BUTTON);
+    }
+  }
+
+  public void selectOverwriteFollowUp(String text) {
+    webDriverHelpers.clickWebElementByText(OVERWRITE_FOLLOW_UP_LABEL, text);
+  }
+
+  public void fillDateOfFollowUpUntil(LocalDate date) {
+    webDriverHelpers.clearAndFillInWebElement(FOLLOW_UP_UNTIL_DATE, formatter.format(date));
+  }
+
+  public void fillFollowUpStatusComment(String source) {
+    webDriverHelpers.clearAndFillInWebElement(FOLLOW_UP_STATUS_TEXT, source);
+  }
+
+  public void selectResponsibleContactOfficer(String vaccine) {
+    webDriverHelpers.selectFromCombobox(RESPONSIBLE_STATUS_OFFICER_COMBOBOX, vaccine);
+  }
+
+  public void fillGeneralComment(String source) {
+    webDriverHelpers.clearAndFillInWebElement(GENERAL_COMMENT, source);
   }
 
   public Contact collectContactData() {
+    Contact contactInfo = getContactInformation();
+    String classification =
+        webDriverHelpers.getCheckedOptionFromHorizontalOptionGroup(CONTACT_CLASSIFICATION);
     String collectedDateOfReport = webDriverHelpers.getValueFromWebElement(REPORT_DATE);
     LocalDate parsedDateOfReport = LocalDate.parse(collectedDateOfReport, formatter);
-    String collectedLastDateOfContact =
-        webDriverHelpers.getValueFromWebElement(DATE_OF_LAST_CONTACT_INPUT);
+    LocalDate parsedDateOfFirst =
+        (webDriverHelpers.isElementVisibleWithTimeout(FIRST_DAY_CONTACT_DATE, 1))
+            ? LocalDate.parse(
+                webDriverHelpers.getValueFromWebElement(FIRST_DAY_CONTACT_DATE), formatter)
+            : null;
```

Comment:
Please remove the logic from here, because inside of the method getValueFromWebElement is already waiting to be present

Reply:
@BojteTamas, please consider that the logic is used because I was extending the method created by @rdutu-vg for creating a contact.

when creating a contact that field is not displayed, because it needs a checkbox action.
i can refactor this the way you've requested but I will have to create a separate method in order for the implementation to work.